org.eclipse.xtext.xtext.ui.tests/src/org/eclipse/xtext/xtext/ui/editor/outline/XtextOutlineTreeProviderTest.java

```diff
@@ -10,37 +10,42 @@
 
 import java.util.List;
 
-import org.eclipse.xtext.ISetup;
 import org.eclipse.xtext.junit4.AbstractXtextTests;
 import org.eclipse.xtext.resource.XtextResource;
+import org.eclipse.xtext.testing.InjectWith;
+import org.eclipse.xtext.testing.XtextRunner;
 import org.eclipse.xtext.ui.editor.model.XtextDocument;
 import org.eclipse.xtext.ui.editor.outline.IOutlineNode;
 import org.eclipse.xtext.ui.editor.outline.impl.OutlineMode;
 import org.eclipse.xtext.util.ITextRegion;
 import org.eclipse.xtext.util.StringInputStream;
-import org.eclipse.xtext.xtext.ui.Activator;
+import org.eclipse.xtext.xtext.ui.XtextUiInjectorProvider;
 import org.junit.Test;
+import org.junit.runner.RunWith;
 
+import com.google.inject.Inject;
 import com.google.inject.Injector;
 
 /**
  * @author Jan Koehnlein - Initial contribution and API
  */
+@RunWith(XtextRunner.class)
+@InjectWith(XtextUiInjectorProvider.class)
 public class XtextOutlineTreeProviderTest extends AbstractXtextTests {
 
+	@Inject
+	private Injector injector;
+
+	@Inject
 	private XtextOutlineTreeProvider treeProvider;
 
+	@Inject
```

Comment:
I'd rather not keep a reference to the document. Please use a `Provider<XtextDocument>` to ensure that two subsequent calls to assertNotException(..) do not interfere with each other.

Reply:
Fixed.